assignments/this.js

```diff
@@ -1,26 +1,73 @@
 /* The for principles of "this";
 * in your own words. explain the four principle for the "this" keyword below.
 *
-* 1. 
-* 2. 
-* 3. 
-* 4. 
+* 1. -->  window Binding
+		  If nothing is bound with 'this' keyword, it will try to go 
+		  to global object window.
+
+* 2. -->  Implicit Binding
+		  Implicit binding for this occurs when dot notation is used to invoke a function.
+
+* 3. -->  Explicit Binding
+		  Explicit binding of this occurs when .call(), .apply(), or .bind() are used on a function.
+		  We call these explicit because you are explicitly passing in a this context to call() or apply().
+
+* 4. -->  new Binding
+          whenever we’ve got a function invoked with the new operator, the this keyword here is bound to 
+          the new object being built by the function constructor.
```

Comment:
I like your answers to these four questions. Clear, concise, and shows that you understand 'this'.